lib/widgets/atoms/profile_header.dart

```diff
@@ -18,52 +19,52 @@ class ProfileHeader extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     final ThemeData theme = Theme.of(context);
+    var backgroundImage = avatarUrl != null
+        ? NetworkImage(avatarUrl!) as ImageProvider<Object>
+        : const AssetImage(Assets.blankAvatar);
+
     return Padding(
-      padding: const EdgeInsets.all(Insets.paddingExtraLarge),
+      padding: const EdgeInsets.fromLTRB(Insets.paddingExtraLarge,
+          Insets.paddingExtraLarge, Insets.paddingExtraLarge, 0),
       child: Row(
         crossAxisAlignment: CrossAxisAlignment.center,
-        mainAxisAlignment: MainAxisAlignment.spaceBetween,
         children: [
+          ClipRRect(
+            borderRadius: BorderRadius.circular(Radii.roundedRectRadiusSmall),
+            child: Image(
+              image: backgroundImage,
+              width: Insets.paddingExtraLarge,
+              height: Insets.paddingExtraLarge,
+              fit: BoxFit.cover,
+            ),
+          ),
           Flexible(
             child: Padding(
-              padding: const EdgeInsets.all(Insets.paddingExtraSmall),
+              padding: const EdgeInsets.all(Insets.paddingSmall),
               child: Text(
                 profileMessage,
                 textAlign: TextAlign.start,
                 softWrap: true,
                 style: theme.textTheme.titleLarge?.copyWith(
                   color: theme.colorScheme.onSurface,
+                  fontWeight: FontWeight.w400,
                 ),
               ),
             ),
           ),
-          Stack(
-            children: [
-              if (profileCompletionPercentage != null)
-                SizedBox(
-                  width: Radii.avatarRadiusSmall * 2,
-                  height: Radii.avatarRadiusSmall * 2,
-                  child: CircularProgressIndicator(
-                    value: profileCompletionPercentage! / 100,
-                  ),
-                ),
-              InkWell(
-                onTap: () {
-                  //TODO(m-rosario): Remove this once the real logout flow is defined by UX
-                  Provider.of<ScaffoldModel>(
-                    context,
-                    listen: false,
-                  ).setTestHomeScaffold();
-                  Scaffold.of(context).openDrawer();
-                },
-                child: CircleAvatar(
-                  radius: Radii.avatarRadiusSmall,
-                  backgroundImage: avatarUrl != null
-                      ? NetworkImage(avatarUrl!) as ImageProvider<Object>
-                      : const AssetImage(Assets.blankAvatar),
-                ),
-              ),
-            ],
+          const Spacer(),
```

Comment:
Wrapping the Text widget with a Flexible solves the overflow issue, but having a Spacer here as well will cause the available space to be divided evenly between the Flexible and the Spacer. Example:

![Screenshot 2023-09-30 at 8 03 24 AM](https://github.com/micromentor-team/mm-flutter-app/assets/25093428/7b6b8317-4b88-4068-91e4-e6ceb486bf7e)

I don't think you need the Spacer here so I would opt for removing it.